packages/requisitions/src/RequestRequisition/api/hooks.ts

```diff
@@ -202,3 +204,56 @@ export const useDeleteSelectedRequisitions = () => {
 
   return deleteAction;
 };
+
+export const useAddFromMasterList = () => {
+  const { error } = useNotification();
+  const queryClient = useQueryClient();
+  const { id: requestId, requisitionNumber } = useRequestFields([
+    'id',
+    'requisitionNumber',
+  ]);
+  const api = useRequestApi();
+  const mutationState = useMutation(api.addFromMasterList, {
+    onSettled: () =>
+      queryClient.invalidateQueries(api.keys.detail(String(requisitionNumber))),
+  });
+
+  const t = useTranslation('distribution');
+  const getConfirmation = useConfirmationModal({
+    title: t('heading.are-you-sure'),
+    message: t('message.confirm-add-from-master-list'),
+  });
+
+  const addFromMasterList = async ({
+    id: masterListId,
+  }: MasterListRowFragment) => {
+    getConfirmation({
+      onConfirm: () =>
+        mutationState.mutate(
+          { masterListId, requestId },
+          {
+            onError: e => {
```

Comment:
The reason to use the `onError` here on the `mutate` fn, rather than on the hook `useMutation`, is because this won't be triggered if the hook is still mounted. If the user navigates away, the toast won't show (is this desired? I think so, but unsure)